app/indexers/work_indexer.rb

```diff
@@ -169,6 +169,22 @@ class WorkIndexer < Kithe::Indexer
           end
         end
       end
+
+      if rec.members.any? {|mem| mem.asset? && mem.english_translation.present?}
+        acc << rec.members.map {|mem| mem.english_translation if mem.asset? }.compact.join(" ")
+      end
+      # Index the transcription into this if we can assume that the work is entirely in English.
+      if rec.members.any? {|mem| mem.asset? && mem.transcription.present?} && rec.language == ['en']
+        acc << rec.members.map {|mem| mem.transcription if mem.asset? }.compact.join(" ")
```

Comment:
you're kind of going through all the members twice -- once to see if it has any member that's relevant, then again to collect all the members that are relevant. Is there a way to do this only once instead?  Might be easier if each element is a separate thing in in the field, no need to join. 

Not sure if this will really make a perf difference but... it might? 

Really not sure how we handle perf issues here.

Reply:
All good points. I didn't want to spend any time optimizing it before you'd had a look, but agree that keeping indexing efficient is very much in our long-interest.

Reply:
Fixed this.